frontend/src/hooks/graph/useGraphEmphasize.ts

```diff
@@ -0,0 +1,73 @@
+import { useTheme } from 'styled-components';
+import * as d3 from 'd3';
+import { useEffect, useRef } from 'react';
+
+export default function useGraphEmphasize(
+  nodeSelector: SVGGElement | null,
+  linkSelector: SVGGElement | null,
+  nodes: any[],
+  links: any[],
+  hoveredNode: string,
+  selectedKey: string,
+) {
+  const theme = useTheme();
+  const styleConstants = useRef({
+    EMPHASIZE_COLOR: theme.COLOR.secondary1,
+    BASIC_COLOR: theme.COLOR.gray1,
+    EMPHASIZE_OPACITY: '1',
+    BASIC_OPACITY: '0.5',
+    EMPHASIZE_STROKE_WIDTH: '0.8px',
+    BASIC_STROKE_WIDTH: '0.5px',
+    EMPHASIZE_STROKE_DASH: 'none',
+    BASIC_STROKE_DASH: '1',
+  });
+
+  useEffect(() => {
+    const styles = styleConstants.current;
+
+    if (nodeSelector === null) return;
+
+    if (hoveredNode === '') {
+      d3.select(nodeSelector).selectAll('text').style('fill-opacity', styles.BASIC_OPACITY);
+    }
+
+    // click/hover된 노드 강조
+    d3.select(nodeSelector)
+      .selectAll('text')
+      .data(nodes)
+      .filter((d) => d.key === selectedKey || d.key === hoveredNode)
+      .style('fill-opacity', styles.EMPHASIZE_OPACITY);
+
+    // click/hover된 노드의 자식 노드들 강조
+    d3.select(nodeSelector)
+      .selectAll('text')
+      .data(nodes)
+      .filter((d) => {
+        const targetList = links
+          .filter((l) => l.source.key === selectedKey || l.source.key === hoveredNode)
+          .map((l) => l.target.key);
+        return targetList.indexOf(d.key) >= 0;
+      })
```

Comment:
우아한 로직이군요.. 멋집니다.

```suggestion
      .filter((d) =>  links
          .filter((l) => l.source.key === selectedKey || l.source.key === hoveredNode)
          .map((l) => l.target.key).includes(d.key);
      })
```

개인적으로는 원소의 존재여부는 includes를 쓰는게 더 가독성이 좋을 것 같아요~

Reply:
includes가 확실히 가독성이 더 좋네요~ 수정하겠습니다.